File: project/jaehyeon-onboarding/src/main/java/fc/icd/jaehyeononboarding/common/model/InputType.java
package fc.icd.jaehyeononboarding.common.model;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum InputType {
    TEXT,
    LONG_TEXT,
    RADIO,
    CHECKBOX;

    @JsonCreator
    public static InputType from(String input) {
        return InputType.valueOf(input.toUpperCase());
    }
}

File: project/sanghyuk-onboarding/src/main/kotlin/fc/innercircle/sanghyukonboarding/form/domain/model/vo/InputType.kt
package fc.innercircle.sanghyukonboarding.form.domain.model.vo

import fc.innercircle.sanghyukonboarding.common.domain.exception.CustomException
import fc.innercircle.sanghyukonboarding.common.domain.exception.ErrorCode

enum class InputType {
    TEXT, // 단답형
    LONG_TEXT, // 장문형
    RADIO, // 라디오 버튼
    CHECKBOX, // 체크박스
    SELECT, // 드롭다운,
    ;

    companion object {
        fun isValidTypeIgnoreCase(type: String): Boolean = InputType.entries.any { it.name == type.uppercase() }
        fun validateOrThrows(type: String) {
            if (!isValidTypeIgnoreCase(type)) {
                throw CustomException(ErrorCode.INVALID_QUESTION_INPUT_TYPE.withArgs(type))
            }
        }

        fun valueOrThrows(type: String): InputType {
            this.validateOrThrows(type)
            return InputType.valueOf(
                type.uppercase()
            )
        }
    }
}
